src/controllers/userController.js

```diff
@@ -0,0 +1,130 @@
+    const token = data.tokens;
+    this.auth.setCredentials(token);
+    const plus = userHelper.getGooglePlusApi(this.auth);
+    const me = await plus.people.get({ userId: 'me' });
+    // const userGoogleId = me.data.id;
+    const email = me.data.emails && me.data.emails.length && me.data.emails[0].value;
+
+    // compare data with database; return success if present else return user not in database
+
+    const user = await models.User.findOne({
+      where: { gmail: email }
+    });
+    if (!user) {
+      return res.status(404).json({ status: 404, message: 'User not found' });
+    }
+    return res.status(200).json({ status: 200, user });
+    } catch (error) {
+      return res.status(599).send({ status: 599, error: 'Please check internet connection' });
+    }
+  }
+
+  //  Create a Linkedin URL and send to the client to log in the user.
+
+  /**
+      * @param {*} req Request object
+      * @param {*} res Response object
+      * @return {json} Returns json object
+      * @memberof User
+      */ 
+  async linkedinUrl(req, res) {
+    this.authurl = Linkedin.auth.authorize(scope);
+    const url = this.authurl;
+    return res.status(200).send({ status: 200, url });
+  }
+
+  // Take the "code" parameter which Linkedin gives us once when the user logs in, get an access token,
+  // then get the user's email.
+  /**
+      * @param {*} req Request object
+      * @param {*} res Response object
+      * @return {json} Returns json object
+      * @memberof User
+      */ 
+  async getLinkedinAccountFromCode(req, res) {
+    Linkedin.auth.getAccessToken(
+      res,
+      req.query.code,
+      req.query.state,
+      async (err, results) => {
+        if (err) return console.error(err);
+        this.accessToken = results.access_token;
+
+        const options = {
+          host: 'api.linkedin.com',
+          path: '/v2/emailAddress?q=members&projection=(elements*(handle~))',
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${this.accessToken}`,
+            'cache-control': 'no-cache',
+            'X-Restli-Protocol-Version': '2.0.0'
+          }
+        };
+
+        const profileRequest = https.request(options, (result) => {
+          let data = '';
+          result.on('data', (chunk) => { data += chunk; });
+          result.on('end', async () => {
+            const profileData = JSON.parse(data);
+            const email = Object.values(profileData.elements[0])[0].emailAddress;
+            console.log('linkedin:', email);
+            const user = await models.User.findOne({
+              where: { linkedin: email }
+            });
+            if (!user) {
+              return res.status(404).json({ status: 404, message: 'User not found' });
+            }
+            return res.status(200).json({ status: 200, user });
+          });
+        });
+        profileRequest.end();
+      }
+    );
+  }
+}
+
+export default new User();
```

Comment:
we already agreed to use static methods and export classes, I think it's better if you try to maintain consistency, however check that it doesn't affect your usage of 'this'

Reply:
Noted